ImporterGA4.php

```diff
@@ -0,0 +1,698 @@
+
+    /**
+     * @var IdMapper
+     */
+    private $idMapper;
+
+    /**
+     * @var int
+     */
+    private $queryCount = 0;
+
+    /**
+     * @var ImportStatus
+     */
+    private $importStatus;
+
+    /**
+     * @var Lock
+     */
+    private $currentLock = null;
+
+    /**
+     * @var string
+     */
+    private $noDataMessageRemoved = false;
+
+    /**
+     * @var ArchiveInvalidator
+     */
+    private $invalidator;
+
+    /**
+     * @var EndDate
+     */
+    private $endDate;
+
+    /**
+     * Whether this is the main import date range or for a reimport range.
+     * @var bool
+     */
+    private $isMainImport = true;
+
+    public function __construct(ReportsProvider                $reportsProvider,
+                                LoggerInterface                $logger,
+                                GoogleGA4GoalMapper            $goalMapper,
+                                GoogleGA4CustomDimensionMapper $customDimensionMapper,
+                                IdMapper                       $idMapper, ImportStatus $importStatus, ArchiveInvalidator $invalidator, EndDate $endDate)
+    {
+        $this->reportsProvider = $reportsProvider;
+        $this->logger = $logger;
+        $this->goalMapper = $goalMapper;
+        $this->customDimensionMapper = $customDimensionMapper;
+        $this->idMapper = $idMapper;
+        $this->importStatus = $importStatus;
+        $this->invalidator = $invalidator;
+        $this->endDate = $endDate;
+    }
+
+    public function setGAClient(BetaAnalyticsDataClient $client)
+    {
+        $this->gaClient = $client;
+    }
+
+    public function setGAAdminClient(AnalyticsAdminServiceClient $client)
+    {
+        $this->gaAdminClient = $client;
+    }
+
+    public function setIsMainImport($isMainImport)
+    {
+        $this->isMainImport = $isMainImport;
+    }
+
+    public function makeSite($propertyId, $timezone = false, $type = Type::ID, $extraCustomDimensions = [], $forceCustomDimensionSlotCheck = false)
+    {
+        if (class_exists(TagManager::class)) {
+            $originalEnableAutoContainerCreation = TagManager::$enableAutoContainerCreation;
+            TagManager::$enableAutoContainerCreation = false;
+        }
+
+        try {
+            $extraCustomDimensions = $this->checkExtraCustomDimensions($extraCustomDimensions);
+
+            $webProperty = $this->gaAdminClient->getProperty($propertyId);
+
+            $startDate = Date::factory($webProperty->getCreateTime()->toDateTime()->getTimestamp())->toString();
+            if (!method_exists(SettingsServer::class, 'isMatomoForWordPress') || !SettingsServer::isMatomoForWordPress()) {
+                $idSite = SitesManagerAPI::getInstance()->addSite(
+                    $siteName = $webProperty->getDisplayName(),
+                    $urls = $type === \Piwik\Plugins\MobileAppMeasurable\Type::ID ? null : [$webProperty->getDisplayName()],
+                    $ecommerce = 1,
```

Comment:
No additional setting to fetch if its Ecommerce since its event based now, it supports ecommerce directly.